From 308d718b0e602819cdf3bf0018b5d4222efca32d Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 06 Jun 2024 15:37:13 +0300
Subject: [PATCH] [FSW-6167] fs1: init pci vf mem bar size

Set the size of VF memory BAR to 512GB for Palad emulator.
Set the alowed sizes of VF memory BAR. Resizable BAR is used
for the VF MEM BAR, so the size is not set statically, but the allowed
sized are set and the actual size is negotiated with the host.

Change-Id: I3de50edadd1e375df75ee6b4f0da8eea42fb7261
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/pldm.c b/arch/fs1/arc/pldm.c
index 732e019..14e4f56 100644
--- a/arch/fs1/arc/pldm.c
+++ b/arch/fs1/arc/pldm.c
@@ -154,7 +154,11 @@
 	}
 }
 
-#define BAR_SIZE_1GB	(SZ_1G - 1)
+#define BAR_SIZE_1GB		(SZ_1G - 1)
+#define VF_RES_BAR_SIZES	(PCIE_DBI_VF_RESBAR_CAP_REG_2_REG_V512GB_M | \
+				 PCIE_DBI_VF_RESBAR_CAP_REG_2_REG_V256GB_M | \
+				 PCIE_DBI_VF_RESBAR_CAP_REG_2_REG_V128GB_M | \
+				 PCIE_DBI_VF_RESBAR_CAP_REG_2_REG_V64GB_M)
 
 static void pldm_pcie_params_init(void)
 {
@@ -179,6 +183,9 @@
 	/* disable shadow registers access */
 	regmap_writel(0, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_AUX_BASE + mmPCIE_AUX_DBI);
 
+	/* set VF resizable BAR #2 allowed size */
+	regmap_writel(VF_RES_BAR_SIZES, IO0_H0_PCIE_DBI(mmPCIE_DBI_VF_RESBAR_CAP_REG_2_REG));
+
 	pldm_pcie_dynamic_init();
 
 	/* disable writing to DBI RO regs */
